operator/controllers/helmrelease_controller.go

```diff
@@ -115,7 +154,54 @@ func (r *HelmReleaseReconciler) onDelete(ctx context.Context, rls shipitv1beta1.
 	return errNotImplemented
 }
 
+func isHelmReleaseNotFound(name string, err error) bool {
+	// dynamic errors can't be directly compared for equality. We use the
+	// error message string, though there's no guarantee it won't change.
+	return strings.Contains(err.Error(), helmerrors.ErrReleaseNotFound(name).Error())
+}
+
 func (r *HelmReleaseReconciler) onUpdate(ctx context.Context, rls shipitv1beta1.HelmRelease) (ctrl.Result, error) {
+	releaseName := rls.Spec.ReleaseName
+
+	_, err := r.helm.ReleaseStatus(releaseName)
+	if err != nil {
+		if isHelmReleaseNotFound(releaseName, err) {
+			return r.install(ctx, rls)
+		}
+		return ctrl.Result{}, errors.Wrapf(err, "failed to get release status for %s", releaseName)
+	}
+
+	return ctrl.Result{}, errNotImplemented
+}
+
+func (r *HelmReleaseReconciler) install(ctx context.Context, rls shipitv1beta1.HelmRelease) (ctrl.Result, error) {
+	chartURI := rls.Spec.Chart.URI()
+	releaseName := rls.Spec.ReleaseName
+
+	chart, err := r.downloader.Download(ctx, chartURI)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to download chart %s", chartURI)
+	}
+
+	// TODO: use the returned response's `Release.Manifest` to watch and
+	// receive events for the k8s resources owned by this chart
+	if _, err := r.helm.InstallReleaseFromChart(chart, r.Namespace, helm.ReleaseName(releaseName)); err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to install release %s using chart %s", releaseName, chartURI)
+	}
+
+	rls.Status.SetCondition(shipitv1beta1.HelmReleaseCondition{
+		Type:    release.Status_PENDING_INSTALL.String(),
+		Message: fmt.Sprintf("installing chart %s", chartURI),
+	})
+
+	if err := r.Update(ctx, &rls); err != nil {
+		r.Log.Info("failed to update HelmRelease status", "release", releaseName, "status", release.Status_PENDING_INSTALL)
+	}
+
+	return ctrl.Result{RequeueAfter: r.GracePeriod}, nil
```

Comment:
I'm not 100% clear on how this works. What will happen if we call this, then the controller pod dies before the grace period?

Reply:
I'm not sure either tbh. I don't know if the controller persists its request queue, but I expect it's probably in memory. 

I'd like to find the answer to that, but I think the best solution would be to eventually implement the "watch resources created by the chart" TODO in this func which completely avoids the need to `RequeueAfter`